### PR TITLE
Prototype support for `--python.target-config`

### PR DESCRIPTION
**Pull request by FFY00**
Early implementation prototype for the static Python installation file format proposed in https://github.com/python/cpython/pull/108483.

### CODE REVIEW

**Comment by FFY00** [comment]
This keeps the Python interpreter discovery logic as-is and simply uses the config file as the highest source of truth, leaving the existent introspection mechanism in place for any information not present in the config.

File: test cases/unit/116 target config/config.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[python]
+# inexistent version number, we set the value to this to test
```
Comment (tristan957) [inline test cases/unit/116 target config/config.toml:2]: non-existent

**Comment by rgommers** [comment]
This PR is superseded by gh-13945 I assume?

**Comment by FFY00** [comment]
Yes.

**Pull request closed without merging**